### PR TITLE
chore: added Pallas compatible comments

### PR DESCRIPTION
**Pull request by tobehn**
I do not know if this is relevant to you.
When i saw Pallas i went through the system to see how it generates the HTML sites. 

I realized it needs the comments in order to provide the small description for each function and struct.
Then i started going through Vib and added the descriptions in *core* everywhere it was missing.

If this is accepted i would go on with adding descriptions.
Let me know if this is wasted time.
Cheers.

### CODE REVIEW

**Review by axtloss — COMMENTED** [review]
This is actually very helpful, thanks for taking the time to create this

but I find the format of the comments to be a bit weird, the comments shouldn't include the function name, and also should be written in the present tense
as an example for the `ChangeWorkingDirectory` function:
```
// Add a WORKDIR instruction to the containerfile
```

**Comment by axtloss** [comment]
Yeah I'm not a fan of the current commenting style, but until we have a set standard i'd keep the format i mentioned for vib

**Comment by tobehn** [comment]
All right I am going to enhance the comments.

File: core/compile.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func CompileRecipe(recipePath string, runtime string, isRoot bool, origGid int,
 	return nil
 }
 
-// compileDocker compiles and builds a Docker image using the provided recipe
+// Build a Docker image using the specified recipe
```
Comment (axtloss) [inline core/compile.go:59]: the docker and podman functions both build oci images, a better comment would be "Build an oci image using the specified recipe through docker"

Comment (axtloss) [reply core/compile.go:59]: same thing for compilePodman

File: core/finalize.go [hunk 2]
```diff
@@ -7,15 +7,14 @@ import (
 	"strings"
 )
 
-// StorageConf represents a configuration for a container storage
+// Holds configuration for storage drivers
```
Comment (axtloss) [inline core/finalize.go:10]: no need for the "holds" part, something like "Configuration for storage drivers" would be better

**Review by axtloss — COMMENTED** [review]
Already good, but one thing is that the return of a function should be seperarated 
instead of "builds xyz and returns abc", it should be
```
builds xyz
returns: abc
```

**Comment by mirkobrombin** [comment]
Small note: Pallas does not care about how you define returns and parameters, it will threat both as a description and always parse both from the AST instead.

**Comment by axtloss** [comment]
> Small note: Pallas does not care about how you define returns and parameters, it will threat both as a description and always parse both from the AST instead.

Yeah I know, but it's easier to read from the description that way, since it makes it easier to spot

**Comment by tobehn** [comment]
> Small note: Pallas does not care about how you define returns and parameters, it will threat both as a description and always parse both from the AST instead.

This is what I also thought.
So in case I write the return into the comment it currently would be rendered into the Pallas generated description where you have a separate section for the returns below.

@mirkobrombin would it be possible to exclude everything from the comment starting with string “return:” from the description in Pallas?
Or even better use it as additional information for the Returns: section.

It then could fit the needs for reading in the code and in the dev documentation.

**Comment by mirkobrombin** [comment]
It is possible, not sure if it is worth it, not sure if we should be opinionated in that side

**Comment by tobehn** [comment]
It also came to my mind it might be difficult to track whether all returns of the function are recorded correctly in the comment.

I'd prefer to use the current auto generation of the functions returns and do not specifically create a section in the comment (=description). In some comments i could describe the returns though, in case they are important.

File: core/compile.go (hunk 1 above)
Comment (tobehn) [reply core/compile.go]: resolved by ddd1e3af7cc494fc31064fa68f5eec14a6b82c24

File: core/finalize.go (hunk 2 above)
Comment (tobehn) [reply core/finalize.go]: resolved by ddd1e3af7cc494fc31064fa68f5eec14a6b82c24

**Review by tobehn — COMMENTED** [review]
solved by ddd1e3af7cc494fc31064fa68f5eec14a6b82c24

**Comment by axtloss** [comment]
> It is possible, not sure if it is worth it, not sure if we should be opinionated in that side

It is, pallas only tells the users the type of the return, not what it returns, which is very important to have

**Comment by axtloss** [comment]
> It also came to my mind it might be difficult to track whether all returns of the function are recorded correctly in the comment.

I don't see how it would be difficult, its not the exact value it returns, but what the return value means, if it's difficult to properly track what the function returns then the function needs to be reworked

File: cmd/build.go [hunk 3]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/vanilla-os/vib/core"
 )
 
+// // Create and return a new build command for the Cobra CLI
```
Comment (axtloss) [inline cmd/build.go:14]:
```suggestion
// Create and return a new build command for the Cobra CLI
```

File: finalize-plugins/shell-final.go [hunk 4]
```diff
@@ -10,14 +10,15 @@ import (
...
 
+// Configuration for a shell command set
 type Shell struct {
 	Name     string   `json:"name"`
 	Type     string   `json:"type"`
 	Commands []string `json:"commands"`
 	Cwd      string   `json:"cwd"`
 }
 
-//export PlugInfo
```
Comment (axtloss) [inline finalize-plugins/shell-final.go:20]: The exports are important, please put the actual comment above them

**Comment by mirkobrombin** [comment]
> > It is possible, not sure if it is worth it, not sure if we should be opinionated in that side
> 
> It is, pallas only tells the users the type of the return, not what it returns, which is very important to have

In that case a proper mapping would be better instead of just removing Returns from the comment

File: api/download.go [hunk 5]
```diff
@@ -180,9 +180,9 @@ func MoveSources(downloadPath string, sourcesPath string, sources []Source, modu
 	return nil
 }
 
-// MoveSource moves a source from the downloads directory to the
-// sources directory, by extracting if a tar archive or moving if a
-// git repository
+// Move or extract a source from the download path to the sources path depending on its type
+// tarballs: extract
+// git repositorys: move
```
Comment (axtloss) [inline api/download.go:185]:
```suggestion
// git repositories: move
```

File: api/finalize-scopes.go [hunk 6]
```diff
@@ -15,6 +15,7 @@ var RUNTIME int32 = 8
 // / Get a read-only filesystem of the Image
 var FS int32 = 16
 
+// Information about the image, recipe, runtime, and file system scope
```
Comment (axtloss) [inline api/finalize-scopes.go:18]:
```suggestion
// Information about the image, recipe, runtime, and file system mountpoint
```

File: cmd/compile.go [hunk 7]
```diff
@@ -68,6 +70,8 @@ func compileCommand(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+// Detect the container runtime by checking system path
+// Returns: runtime name or an empty string if none is found
```
Comment (axtloss) [inline cmd/compile.go:74]:
```suggestion
// Returns: runtime name or an empty string if no runtime is found
```

File: cmd/compile.go [hunk 8]
```diff
@@ -68,6 +70,8 @@ func compileCommand(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+// Detect the container runtime by checking system path
```
Comment (axtloss) [inline cmd/compile.go:73]:
```suggestion
// Detect the container runtime by checking the system path
```

File: core/build.go [hunk 9]
```diff
@@ -35,7 +37,7 @@ func RestoreWorkingDirectory(workdir string, containerfile *os.File) error {
 	return nil
 }
 
-// BuildRecipe builds a Containerfile from a recipe path
+// Load and build the recipe
```
Comment (axtloss) [inline core/build.go:40]:
```suggestion
// Load and build a Containerfile out of the given recipe
```

File: core/build.go [hunk 10]
```diff
@@ -63,8 +65,7 @@ func BuildRecipe(recipePath string) (api.Recipe, error) {
 	return *recipe, nil
 }
 
-// BuildContainerfile builds a Containerfile from a recipe
-// and a list of modules commands
+// Generate the Containerfile from the recipe
```
Comment (axtloss) [inline core/build.go:68]:
```suggestion
// Generate a Containerfile from the recipe
```

File: finalize-plugins/genimage.go [hunk 11]
```diff
@@ -20,6 +21,8 @@ type Genimage struct {
 	Outputpath   string `json:"outputpath"`
 }
 
+// Provide information about a plugin as a JSON string
```
Comment (axtloss) [inline finalize-plugins/genimage.go:24]:
```suggestion
// Provide information about the plugin as a JSON string
```

File: finalize-plugins/shell-final.go [hunk 12]
```diff
@@ -10,13 +10,16 @@ import (
 	"strings"
 )
 
+// Configuration for a shell command set
```
Comment (axtloss) [inline finalize-plugins/shell-final.go:13]:
```suggestion
// Configuration for a set of shell commands
```

File: finalize-plugins/shell-final.go [hunk 13]
```diff
@@ -27,17 +30,24 @@ func PlugInfo() *C.char {
 	return C.CString(string(pluginjson))
 }
 
+// Provide the plugin scope as a bitmask
```
Comment (axtloss) [inline finalize-plugins/shell-final.go:33]: I'm not sure if its important to specify that its provided as a bitmask, though i'm open to hear your reasoning if you think its important

**Review by axtloss — COMMENTED** [review]
Some small nitpicks that i came across while going through the changes, starting to look pretty nice though

File: core/build.go (hunk 9 above)
Comment (tobehn) [reply core/build.go:40]: I'd prefer `// Load and build a Containerfile from the specified recipe`

Comment (axtloss) [reply core/build.go:40]: yeah that would work too

File: finalize-plugins/shell-final.go (hunk 13 above)
Comment (tobehn) [reply finalize-plugins/shell-final.go:33]: `as a bitmask` is no important information. I totally agree.

**Comment by axtloss** [comment]
@tobehn any update on this? is it ready to merge?

**Comment by tobehn** [comment]
> @tobehn any update on this? is it ready to merge?

From my point of view the comments are done.

**Comment by axtloss** [comment]
Alright, could you squash your commits into one so that I can merge it?

**Comment by tobehn** [comment]
oh gosh, it looks to me like i messed it up. 
I've never done a squashing before. Can you tell if i have done it right?

**Comment by axtloss** [comment]
Yep, everything's great and how it's supposed to be.
Thanks for going through the effort and commenting everything!

**Review by axtloss — APPROVED** [review]

**Pull request merged**